lib/init-lib/build.zig

```diff
@@ -15,6 +15,12 @@ pub fn build(b: *std.Build) void {
     // set a preferred release mode, allowing the user to decide how to optimize.
     const optimize = b.standardOptimizeOption(.{});
 
+    // This creates a module that can be imported into a zig project by its given name.
+    // The returned *Module can be imported by using std.Build.LibExeObjStep.addModule.
```

Comment:
LibExeObjStep no longer exists. Also, I think this should actually show the usage rather than just explaining it in a comment.

Reply:
Oops, I'll change that. I based that off of autodoc's link.

Should I add a consumer file for basic usage? (Creating a separate exe source file) or should I add a comment with the code required to consume the module? 
```zig 
// This creates a module that can be imported into a zig project / build step by its given name.
// The returned module can be consumed within the same package for different Step.Compiles
//const mod = b.addModule("$", .{
_ = b.addModule("$", .{
    .source_file = .{ .path = "src/main.zig" },
});

const lib = b.addStaticLibrary(.{
    .root_source_file = .{ .path = "src/main.zig" },
});

// This is how mod can be consumed by lib
lib.addModule("mod", mod);

// Use std.build.dependency() to add in a package from build.zig.zon 
// (for use with official package manager)
//const package_mod = b.dependency("package", .{}).module("mod_name");

// This adds module "mod_name" to exe as "pm_lib", allowing import with @import("lib")
//exe.addModule("pm_lib", package_mod);
```
or
```zig
// build.zig
const mod = b.addModule("$", .{
    .source_file = .{ .path = "src/main.zig" },
});

const exe = b.addExecutable(.{
    .root_source_file = .{ .path = "src/executable.zig" },
});
exe.addModule("lib", mod);

// executable.zig
const std = @import("std");
const lib = @import("lib");

pub fn main() void {
    std.debug.print("{d}\n", .{lib.add(9, 10)});
}

Reply:
I went with the "add another file" route. Let me know if this poses an issue